util/encoding/encoding_test.go

```diff
@@ -643,6 +646,60 @@ func TestEncodeDecodeNull(t *testing.T) {
 	}
 }
 
+func TestEncodeDecodeTime(t *testing.T) {
+	zeroTime := time.Unix(0, 0)
+
+	zeroTime = zeroTime.In(time.FixedZone("", -7*60*60))
+
+	// test cases are negative, increasing, duration offsets from the
+	// zeroTime. The positive, increasing, duration offsets are automatically
+	// genarated below.
+	testCases := []string{
```

Comment:
Other things to test: precision of units smaller than 1ms, non-zero times with zero fractional parts.